domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java

```diff
@@ -82,9 +82,10 @@
     private AccessIdentityResourceDefinition(AbstractAddStepHandler add) {
         super(new Parameters(PATH_ELEMENT, DomainManagementResolver.getResolver("core.identity"))
                 .setAddHandler(add)
-                .setRemoveHandler(new ReloadRequiredRemoveStepHandler(MANAGEMENT_IDENTITY_RUNTIME_CAPABILITY))
+                .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE)
                 .setAddRestartLevel(OperationEntry.Flag.RESTART_NONE)
                 .setRemoveRestartLevel(OperationEntry.Flag.RESTART_RESOURCE_SERVICES)
+                .setCapabilities(MANAGEMENT_IDENTITY_RUNTIME_CAPABILITY)
```

Comment:
is your change equivalent?

Capability added to the ReloadRequiredRemoveStepHandler will be handled in org.jboss.as.controller.AbstractRemoveStepHandler#recordCapabilitiesAndRequirements that does a whole lot more that when you set it on the resource definition where it is handled in org.jboss.as.controller.SimpleResourceDefinition#registerCapabilities.

I agree with your change in principle as I think the capabilities belongs to the resource and not to its add/remove operation. However, are we missing some process from org.jboss.as.controller.AbstractRemoveStepHandler#recordCapabilitiesAndRequirements in doing so?

Reply:
> is your change equivalent?

Yes - the resulting behavior is exactly the same.
See: https://github.com/wildfly/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/AbstractRemoveStepHandler.java#L185

Reply:
ah, right.

So my question is not specific to your PR but more generally why have we 2 different ways to specify capabilities (either in add/remove handler or in the resource definition)?
In any case, your PR is fine wrt that issue

Reply:
@jmesnil That's just a matter of code history; in the initial cap/req code there was no MRR.getCapabilities(); the relevant caps were stored in the 'capabilities' field in the add/remove handlers.